src/test/groovy/ratpack/zipkin/ServerTracingModuleSpec.groovy

```diff
@@ -492,4 +496,79 @@ class ServerTracingModuleSpec extends Specification {
             assertThat(span.name).isEqualTo("say/:message")
 	}
+	def 'Should allow span tag customization from request'() {
+		given:
+            def app = GroovyEmbeddedApp.of { server ->
+                server.registry(Guice.registry { binding ->
+                    binding.module(ServerTracingModule.class, { config ->
+                        config
+                            .serviceName("embedded")
+                            .sampler(Sampler.ALWAYS_SAMPLE)
+                            .spanReporter(reporter)
+							.requestTagCustomizer(new RequestTagCustomizer() {
+							@Override
+							List<Pair<String, String>> tags(final ServerRequest request) {
+								return Collections.singletonList(Pair.pair("foo","bar"))
+							}
+						})
+                    }) }).handlers {
+                    chain ->
+                        chain.get("/test", new Handler() {
+                            @Override
+                            void handle(final Context ctx) throws Exception {
+                                ctx.response.send("yo!")
+                            }
+                        })
+                }
+            }
+		when:
+            app.test { t ->
+                t.get("/test")
+            }
+		then:
+            assertThat(reporter.getSpans()).isNotEmpty()
+            def span = reporter.getSpans().first()
+			def tag = span.binaryAnnotations.stream().filter{ba -> ba.key == "foo"}.findFirst()
+			assertThat(tag).isPresent()
+			tag.get().value == "bar".getBytes()
+	}
+
+	def 'Should allow span tag customization from response'() {
+		given:
+            def app = GroovyEmbeddedApp.of { server ->
+                server.registry(Guice.registry { binding ->
+                    binding.module(ServerTracingModule.class, { config ->
+                        config
+                            .serviceName("embedded")
+                            .sampler(Sampler.ALWAYS_SAMPLE)
+                            .spanReporter(reporter)
+                            .responseTagCustomizer(new ResponseTagCustomizer() {
+							@Override
+							List<Pair<String, String>> tags(final Response response, final PathBinding pathBinding) {
+								return pathBinding.getTokens()
+										.entrySet().stream()
+										.map{ entry -> Pair.of(entry.key, entry.value) }.collect(Collectors.toList())
+							}
+						})
+                    }) }).handlers {
+                    chain ->
+                        chain.get("say/:msg", new Handler() {
+                            @Override
+                            void handle(final Context ctx) throws Exception {
+                                ctx.response.send("yo!")
+                            }
+                        })
+                }
+            }
+		when:
+            app.test { t ->
+                t.get("/say/hello")
+            }
+		then:
+            assertThat(reporter.getSpans()).isNotEmpty()
+            def span = reporter.getSpans().first()
```

Comment:
Ps switching your test span reporter to use v2 format will make this easier to read.

Reply:
@adriancole I guess that would be something that depends on the upgrade to v2 spans (#40)?

Reply:
@adriancole I rebased after #40 was merged and re-worked the tests a bit. Also tried to flesh out the PR description a bit.